persistence/scaladsl/src/test/scala/com/lightbend/lagom/scaladsl/persistence/TestEntity.scala

```diff
@@ -67,10 +66,59 @@ object TestEntity {
   final case class AfterRecovery(state: State)
 }
 
-class TestEntity @Inject() (system: ActorSystem, probe: Option[ActorRef] = None)
+// NOTE needs to be registered in actor system config to be picked up like this:
+// lagom.serialization.play-json.serialization-registry =
+//   "com.lightbend.lagom.scaladsl.persistence.TestEntitySerializerRegistry"
+class TestEntitySerializerRegistry extends SerializerRegistry {
+  import TestEntity._
+  import play.api.libs.json._
+
+  implicit val modeFormat = Format[Mode](
+    Reads[Mode] {
+      case JsString("append")  => JsSuccess(Mode.Append)
+      case JsString("prepend") => JsSuccess(Mode.Prepend)
+      case js                  => JsError(s"unknown mode js: $js")
+    }, Writes[Mode] {
+      case Mode.Append  => JsString("append")
+      case Mode.Prepend => JsString("prepend")
+    }
+  )
+
+  def emptySingletonFormat[A](singleton: A) = Format[A](
+    Reads[A](_ => JsSuccess(singleton)),
+    Writes[A](_ => JsObject(Seq.empty))
+  )
+
+  override def serializers: Seq[Serializers[_]] = Seq(
+    // commands
+    Serializers(Json.format[Add]),
+    Serializers(Json.format[ChangeMode]),
+    Serializers(emptySingletonFormat(Get)),
+    Serializers(emptySingletonFormat(UndefinedCmd)),
+    Serializers(emptySingletonFormat(GetAddress)),
+
+    // events
+    Serializers(Json.format[Appended]),
+    Serializers(Json.format[Prepended]),
+    Serializers(emptySingletonFormat(InPrependMode)),
+    Serializers(emptySingletonFormat(InAppendMode)),
+
+    // state
+    Serializers(Json.format[State])
+  )
```

Comment:
yes, I don't see a way around that runtime testing is needed to find missing serializers. The `PersistentEntityTestDriver` will find it for everything that is exercised. (javadsl is not better, since you can forget to extend `Jsonable`)

Is it possible to find all concrete subclasses with reflection? Then it could do additional checking automatically.

It would be nice to use a convention that the serializers are defined in the companion objects of the `Event`, `Command` and `State`. Then you at least have them in the same place as the actual classes.

```scala
class MyServiceSerializerRegistry extends SerializerRegistry {
  override def serializers: Seq[Serializers[_]] =
    TestEntity.serializers ++
    AnotherEntity.serializers
}

object TestEntity {
  def serializers: Seq[Serializers[_]] =
    TestEntity.Evt.serializers ++
    TestEntity.Command.serializers ++
    TestEntity.State.serializers
}

object AnotherEntity {
  def serializers: Seq[Serializers[_]] = ???
}
```